### ISSUE TITLE
"New Preview Prototype" tab

### ISSUE THREAD

**Issue by colinsutton** [eclipse-birt/birt#1594]
With the birt-report-designer-all-in-one-4.15.0-202403111050-macosx.cocoa.aarch64 an undocumented "New Preview Prototype" tab appears in the UI. 
<img width="730" alt="screenshot_789" src="https://github.com/eclipse-birt/birt/assets/18058891/eee41bb2-d70b-4461-b19c-cdd23eb623cb">

I can see code was added in 2007, but the tab did not appear in release 4.9, for example.

**Comment by speckyspooky** [comment]
What is here the question or the issue?
It is a prototype of an HTML-preview. I cannot say when it was activated. But the tab is available already with BIRT 4.14

**Comment by colinsutton** [comment]
My issue was just that it is undocumented - is it useful? How? It seems to be a tab with a window that doesn't accept input.
I was busy when 4.14 was released, didn't see that one.

**Comment by speckyspooky** [comment]
Correct, the preview doesn't accept inputs and not all elements will be displayed according to the standard BIRT-viewer-HTML.
This prototype is a try to get faster the first impressions of the report results based on the internal eclipse browser.

I don't know which browser engine was used internally.
And from my point of view it is really an "early prototype" due to the missing options of the viewer.
And I'm with you that a small documentation would help to understand clearly the intention and also the next steps.
(I'm not the author of it.)

To decide whether it is useful, it is dependend on your use cases.
I work like before with the standard viewer-preview (HTML-preview).

**Comment by colinsutton** [comment]
..."not all elements will be displayed"...
 I don't see anything displayed. I just noticed there is an error when I select that tab: "Unhandled event loop exception"
[Unhandled event loop exception.txt.zip](https://github.com/eclipse-birt/birt/files/14596726/Unhandled.event.loop.exception.txt.zip)

**Comment by colinsutton** [comment]
Here's a sample design that shows the problem. It accesses a local time_zone.csv for its data source, then displays the first three columns from the dataset in view report html. On selecting the 'New Preview Prototype' the error occurs.
[Archive.zip](https://github.com/eclipse-birt/birt/files/14597001/Archive.zip)

**Comment by speckyspooky** [comment]
The problem is solved with the Ticket #1588.
There was a change coming during the eclipse-platform and I fixed that issue on BIRT side.
The nightly build shows currently more details on this tab.

But at the end it is a prototype and I don't know who developer will support it.
It is hard to have 2 previews - in my opinion it makes no sense. Therefore I ignore the "prototype tab"

**Comment by hvbtup** [comment]
> Therefore I ignore the "prototype tab"
Me too. I am usually not interested in HTML output. Most of the time I use "Run report as PDF".
And sometimes "... as HTML", because during development I like the fact that the error messages are rendered in red at the bottom of the report, including stack trace. This is more useful than the "Problems" register, where it's hard to get at the details.
